### ISSUE TITLE
ZIP64 files containing more than 65,535 entries only report 65,535 entries

### ISSUE THREAD

**Issue by gdubois** [nmoinvaz/minizip#457]
When used on seemingly compliant ZIP64 zip files that contain over 65,535 entries, `mz_zip_get_number_entry` returns the 16 bit value from the central directory instead of the value from the ZIP64 extension.

In the `mz_zip.c`  zip64 section, the correct value is read to `number_entry`
```if (err == MZ_OK)
                    err = mz_stream_read_uint64(zip->stream, &number_entry);
```
but it seems that `number_entry` is never subsequently assigned to `zip->number_entry` which thus keeps its value of 65,535

**Comment by nmoinvaz** [comment]
It maybe be that the next line:
```
if (number_entry == UINT32_MAX)
    zip->number_entry = number_entry_cd64;
```
Should be:
```
if (number_entry_cd == UINT32_MAX)
    zip->number_entry = number_entry_cd64;
```
Can you check to see if that works for you?

**Comment by gdubois** [comment]
Thanks for the quick feedback and for maintaining this lib in general.

I fixed the issue by changing this line 
```err = mz_stream_read_uint64(zip->stream, &number_entry);``` to ```err = mz_stream_read_uint64(zip->stream, &zip->number_entry);```
and also 
```if (number_entry == UINT32_MAX)
                    zip->number_entry = number_entry_cd64;```
to
```if (zip->number_entry != number_entry_cd64)
                    err = MZ_FORMAT_ERROR;```
for consistency with your code a bit higher up handling the non zip64 number of entries; where you  handle a mismatch between "the number of entries in the central dir" and "the number of entries in the central dir on this disk".

Your proposition will unfortunately not work because:
- the comparison should be with UINT16_MAX (according to the specs number_entry_cd is a 2 byte value containing 0xFFFF in that case)
- in this code block you have already checked that number_entry_cd == UINT16_MAX anyway

My suggestion above does work on all the zip64 and non zip64 files I have tested it on. I can make a merge request if you are happy with it.
